driver-embedded/src/main/com/mongodb/embedded/client/MongoDBCAPIHelper.java

```diff
@@ -37,6 +37,9 @@
     private static final Logger LOGGER = Loggers.getLogger("embedded.server");
     private static final String NATIVE_LIBRARY_NAME = "mongo_embedded_capi";
     private static volatile MongoDBCAPI mongoDBCAPI;
+    private static volatile Pointer status;
```

Comment:
While it's not explicitly stated in libmongocapi.h documentation, I don't see how it could be safe to use the same status pointer concurrently in multiple threads. Each libmongocapi operation that takes a status pointer can modify it with error information, and I doubt that a single one can handle more than one at a time.

So I think that each client pointer has to have its own status pointer associated with it.  And then some number of others for the higher level calls like init and fini.